src/dynd/types/tuple_type.cpp

```diff
@@ -58,7 +58,7 @@ ndt::tuple_type::tuple_type(type_id_t type_id, const nd::array &field_types, fla
   m_arrmeta_offsets.flag_as_immutable();
 }
 
-ndt::tuple_type::tuple_type(const nd::array &field_types, bool variadic)
+ndt::tuple_type::tuple_type(std::vector<type> field_types, bool variadic)
     : tuple_type(tuple_id, field_types, type_flag_none, true, variadic)
```

Comment:
`std::move(field_types)` here too, and also anywhere else the by-value std::vector<type> parameter is.